src/components/plots/barPlot.ts

```diff
@@ -237,38 +237,40 @@ export module Plots {
     }
 
     /**
-     * Gets the bar under the given pixel position (if [xValOrExtent]
-     * and [yValOrExtent] are {number}s), under a given line (if only one
-     * of [xValOrExtent] or [yValOrExtent] are {Extent}s) or are under a
-     * 2D area (if [xValOrExtent] and [yValOrExtent] are both {Extent}s).
-     *
-     * @param {number | Extent} xValOrExtent The pixel x position, or range of x values.
-     * @param {number | Extent} yValOrExtent The pixel y position, or range of y values.
-     * @returns {D3.Selection} The selected bar, or null if no bar was selected.
+     * Gets the {Plots.PlotData} that correspond to the given pixel position.
+     * 
+     * @param {Point} p The provided pixel position as a {Point}
+     * @return {Plots.PlotData} The plot data that corresponds to the {Point}.
      */
-    public getBars(xValOrExtent: number | Extent, yValOrExtent: number | Extent): D3.Selection {
-      if (!this._isSetup) {
-        return d3.select();
-      }
-
-      // currently, linear scan the bars. If inversion is implemented on non-numeric scales we might be able to do better.
-      var bars = this._datasetKeysInOrder.reduce((bars: any[], key: string) =>
-        bars.concat(this._getBarsFromDataset(key, xValOrExtent, yValOrExtent))
-      , []);
+    public plotDataAt(p: Point) {
+      return this._getPlotData(p.x, p.y);
+    }
 
-      return d3.selectAll(bars);
+    /**
+     * Gets the {Plots.PlotData} that correspond to a given xRange/yRange
+     * 
+     * @param {Extent} xRange The specified range of x values
+     * @param {Extent} yRange The specified range of y values
+     * @return {Plots.PlotData} The plot data that corresponds to the ranges
+     */
+    public plotDataIn(xRange: Extent, yRange: Extent) {
```

Comment:
To correlate with the other `plotData` API should we include a datasets parameter?  Possibly out of scope.